src/bindgen/builder.rs

```diff
@@ -394,6 +394,29 @@ impl Builder {
 
         result.source_files.extend_from_slice(self.srcs.as_slice());
 
+        // Replace associated types in function signatures
+        for func in result.functions.iter_mut(){
+            // Check return type
+            if let Some(key) = &func.ret.1{
+                let concrete_type = result.assoc_types.get(key);
+                if let Some(ty) = concrete_type{
+                    func.ret.0 = ty.clone();
+                }
+            }
+
+            // Check arguments type
+            for i in 0..func.args.len(){
+                let arg = &mut func.args[i];
+                if let Some(key) = &arg.assoc_type{
+                    let concrete_type = result.assoc_types.get(key);
+                    if let Some(ty) = concrete_type{
+                        arg.ty = ty.clone();
+                    }
+                }
+            }
```

Comment:
this is not how we write Rust code. Change it to this:
```suggestion
            for arg in &mut func.args[i]{
                if let Some(key) = &arg.assoc_type{
                    let concrete_type = result.assoc_types.get(key);
                    if let Some(ty) = concrete_type{
                        arg.ty = ty.clone();
                    }
                }
            }
```